### PR TITLE
Add build-chain doc

### PR DESCRIPTION
**Pull request by 0xmichalis**

### CODE REVIEW

**Comment by adellape** [comment]
@kargakis Thanks. I'll wait for the code PR to merge before merging the docs here.

BTW, the new file won't show up in the TOC until it's added to the `_build_cfg.yml` tree.

**Comment by kargakis** [comment]
> BTW, the new file won't show up in the TOC until it's added to the _build_cfg.yml tree.

@adellape thanks for pointing this out, I have updated the YAML file with the appropriate changes.

File: using_openshift/output_deps.adoc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+Build dependency tree for the specified image repository with the specified tag in the current namespace and output it in JSON
+----
+$ openshift ex build-chain [image-repository]:[tag]
+----
+
+Build and output the dependency tree for the image repository in the specified namespace in DOT syntax
+----
+$ openshift ex build-chain [namespace]/[image-repository] -o dot
```
Comment (bparees) [inline using_openshift/output_deps.adoc]: why are we using namespace/imagerepo syntax here when we use "-n namespace" to specify the namespace in all our other commands?

File: using_openshift/output_deps.adoc [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+:toc: macro
+:toc-title: 
+
+toc::[]
+
+== Overview
+`build-chain` is an experimental command-line utility for OpenShift v3. It outputs the build dependency tree of a given image repository
+in JSON (default), DOT, or AST syntax. The user can specify an image repository name (and optionally a namespace and a tag) to build its
+dependency tree, call the tool without any arguments to build all dependency trees in the current namespace, or specify the `--all` flag to build all dependency trees in all available namespaces.
+
```
Comment (bparees) [inline using_openshift/output_deps.adoc]: This overview needs a little work, it should explain what  build dependency is.  eg:  This tool will start at the imagerepo:tag specified and determine what Builds will be triggered (due to ImageChangeTriggers on the BuildConfig) when this imagerepo:tag is updated.  Further, it analyzes the output of those Builds to determine if they in turn update an imagerepo:tag.  If they do, the tool continues to follow the dependency tree.  It then outputs the total graph of what imagerepo:tags will be impacted by an update to the top level imagerepo:tag.

File: using_openshift/output_deps.adoc (hunk 1 above)
Comment (bparees) [inline using_openshift/output_deps.adoc]: also needs to indicate what happens if you do not specifiy a tag (assumes "latest" right?)

File: using_openshift/output_deps.adoc [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+----
+$ openshift ex build-chain [image-repository]:[tag]
+----
+
+Build and output the dependency tree for the image repository in the specified namespace in DOT syntax
+----
+$ openshift ex build-chain [namespace]/[image-repository] -o dot
+----
+
+Build dependency trees for all image repositories in the current namespace and output them in JSON
```
Comment (bparees) [inline using_openshift/output_deps.adoc]: all image repositories and all tags of those imagerepositories.

Comment (bparees) [inline using_openshift/output_deps.adoc]: (at least I assume?)

File: using_openshift/output_deps.adoc (hunk 1 above)
Comment (kargakis) [inline using_openshift/output_deps.adoc]: I was probably carried away with the parsing of DockerImageReferences...

Comment (kargakis) [inline using_openshift/output_deps.adoc]: > also needs to indicate what happens if you do not specifiy a tag (assumes "latest" right?)

Yep

File: using_openshift/output_deps.adoc (hunk 3 above)
Comment (kargakis) [inline using_openshift/output_deps.adoc]: > all image repositories and all tags of those imagerepositories.

Yep

**Comment by kargakis** [comment]
@bparees are you ok with this now?

File: using_openshift/output_deps.adoc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ 
+== Examples
+
+
+Build dependency tree for the specified image repository with the specified tag in the current namespace and output it in JSON
+----
+$ openshift ex build-chain [image-repository]:[tag]
+----
+
+Build and output the dependency tree for the specified image repository in 'testing' namespace using DOT syntax
```
Comment (bparees) [inline using_openshift/output_deps.adoc]: this will use the "latest" tag since none is specified, right?  should indicate that.

**Comment by bparees** [comment]
@kargakis one minor comment and then lgtm.

**Comment by kargakis** [comment]
@adellape this can now get merged

**Comment by adellape** [comment]
@kargakis Thanks, merging.

@tpoitras FYI per https://trello.com/c/QeyUIKd8.

**Pull request merged**